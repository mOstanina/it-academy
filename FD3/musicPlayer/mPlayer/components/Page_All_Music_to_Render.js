//Page_All_Music_to_Render
import React from 'react';
import PropTypes, { func } from 'prop-types';

import Track from '../components/track';
//import "./Page_All_Music.css"


import "../pages/Page_All_Music.css"
class Page_All_Music_to_Render extends React.PureComponent {
    static propTypes = {
        listOfAllSongs: PropTypes.arrayOf(
            PropTypes.shape({
                code: PropTypes.number.isRequired,
                isInList: PropTypes.bool.isRequired,
                groupName: PropTypes.string.isRequired,
                songName: PropTypes.string.isRequired,
                url: PropTypes.string.isRequired,
            })
        ),
        filteredString: PropTypes.func,
        clear: PropTypes.func,
        showAll: PropTypes.func,
        setActiv: PropTypes.func,
        setBlock: PropTypes.func,
        filtered: PropTypes.string,
        workMode: PropTypes.string,
        cardMode: PropTypes.string,
        userPlayList: PropTypes.array,
        disab: PropTypes.bool,
        toAddSong: PropTypes.func,
    };

    render() {

        let listAllSongs = this.props.listOfAllSongs.map((song, i) => {
            return <Track key={song.code} info={song} workMode={this.props.workMode} cardMode={this.props.cardMode} userPlayList={this.props.userPlayList} disab={this.props.userPlayList.includes(song.code)} cbToAddSong={this.props.toAddSong} />
        })
        return (

            <div className="pageContainerOfMainPage">
                <div className='btn_All_Music'>
                    <div className="input_text">
                        <input type="text" value={this.props.filtered} onChange={this.props.filteredString} />
                        <input type="button" value="очистить поиск" onClick={this.props.clear} />
                    </div>
                    <div className="input_btn">
                        <input type="button" value="показать все" onClick={this.props.showAll} />
                        <input type="button" value="показать недобавленные" onClick={this.props.setActiv} />
                        <input type="button" value="показать добавленные" onClick={this.props.setBlock} />
                    </div>
                </div>
                {listAllSongs}
            </div>
        );

    };
}
export default Page_All_Music_to_Render;