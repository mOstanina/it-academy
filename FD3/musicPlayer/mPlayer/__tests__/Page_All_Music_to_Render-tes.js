"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import Page_All_Music_to_Render from '../components/Page_All_Music_to_Render';

test('работа Page_All_Music_to_Render', () => {

  // создаём тестовую версию компонента
  let list = [
    { code: 44, isInList: true, groupName: "Status Quo", songName: "In The Army Now", url: "../songs/Status Quo - In The Army Now.mp3" },
    { code: 45, isInList: true, groupName: "Stereophonics", songName: "Maybe Tomorrow", url: "../songs/Stereophonics - Maybe Tomorrow.mp3" },
    { code: 46, isInList: false, groupName: "System Of A Dow", songName: "Lonely Day", url: "../songs/System Of A Down - Lonely Day.mp3" },
    { code: 47, isInList: true, groupName: "The Beatles", songName: "Come Together", url: "../songs/The Beatles - Come Together.mp3" },
  ];
  let userPL=[45,48];
  let filteredStr=""

  const component = renderer.create(
    <Page_All_Music_to_Render
      listOfAllSongs={list}
      filtered={filteredStr}
      workMode={"allSongs"}
      cardMode={"shortMode"}
      userPlayList={userPL}
    />
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('работа Page_All_Music_to_Render (выбрана сортировка по букве T)', () => {

  // создаём тестовую версию компонента
  let list = [
    { code: 44, isInList: true, groupName: "Status Quo", songName: "In The Army Now", url: "../songs/Status Quo - In The Army Now.mp3" },
    { code: 45, isInList: true, groupName: "Stereophonics", songName: "Maybe Tomorrow", url: "../songs/Stereophonics - Maybe Tomorrow.mp3" },
    { code: 46, isInList: false, groupName: "System Of A Dow", songName: "Lonely Day", url: "../songs/System Of A Down - Lonely Day.mp3" },
    { code: 47, isInList: true, groupName: "The Beatles", songName: "Come Together", url: "../songs/The Beatles - Come Together.mp3" },
  ];
  let userPL=[45,48];
  let filteredStr="T"

  const component = renderer.create(
    <Page_All_Music_to_Render
      listOfAllSongs={list}
      filtered={filteredStr}
      workMode={"allSongs"}
      cardMode={"shortMode"}
      userPlayList={userPL}
    />
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

