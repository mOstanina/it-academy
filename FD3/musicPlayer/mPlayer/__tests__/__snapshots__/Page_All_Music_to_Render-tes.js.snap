// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Page_All_Music_to_Render (выбрана сортировка по букве T) 1`] = `
<div
  className="pageContainerOfMainPage"
>
  <div
    className="btn_All_Music"
  >
    <div
      className="input_text"
    >
      <input
        onChange={undefined}
        type="text"
        value="T"
      />
      <input
        onClick={undefined}
        type="button"
        value="очистить поиск"
      />
    </div>
    <div
      className="input_btn"
    >
      <input
        onClick={undefined}
        type="button"
        value="показать все"
      />
      <input
        onClick={undefined}
        type="button"
        value="показать недобавленные"
      />
      <input
        onClick={undefined}
        type="button"
        value="показать добавленные"
      />
    </div>
  </div>
  <div
    className="container_track"
  >
    <div
      className="shortMode_track"
    >
      <div
        className="for_span_track"
      >
         
        <span
          className="songN"
        >
          Status Quo
           
          -In The Army Now
        </span>
         
      </div>
      <div
        className="for_audio_track"
      >
           
        <audio
          controls={true}
          src="../songs/Status Quo - In The Army Now.mp3"
        />
      </div>
      <div
        className="for_input_track"
      >
          
        <input
          disabled={false}
          onClick={[Function]}
          type="button"
          value="add to my play-list"
        />
      </div>
    </div>
  </div>
  <div
    className="container_track"
  >
    <div
      className="shortMode_track"
    >
      <div
        className="for_span_track"
      >
         
        <span
          className="songN"
        >
          Stereophonics
           
          -Maybe Tomorrow
        </span>
         
      </div>
      <div
        className="for_audio_track"
      >
           
        <audio
          controls={true}
          src="../songs/Stereophonics - Maybe Tomorrow.mp3"
        />
      </div>
      <div
        className="for_input_track"
      >
          
        <input
          disabled={true}
          onClick={[Function]}
          type="button"
          value="add to my play-list"
        />
      </div>
    </div>
  </div>
  <div
    className="container_track"
  >
    <div
      className="shortMode_track"
    >
      <div
        className="for_span_track"
      >
         
        <span
          className="songN"
        >
          System Of A Dow
           
          -Lonely Day
        </span>
         
      </div>
      <div
        className="for_audio_track"
      >
           
        <audio
          controls={true}
          src="../songs/System Of A Down - Lonely Day.mp3"
        />
      </div>
      <div
        className="for_input_track"
      >
          
        <input
          disabled={false}
          onClick={[Function]}
          type="button"
          value="add to my play-list"
        />
      </div>
    </div>
  </div>
  <div
    className="container_track"
  >
    <div
      className="shortMode_track"
    >
      <div
        className="for_span_track"
      >
         
        <span
          className="songN"
        >
          The Beatles
           
          -Come Together
        </span>
         
      </div>
      <div
        className="for_audio_track"
      >
           
        <audio
          controls={true}
          src="../songs/The Beatles - Come Together.mp3"
        />
      </div>
      <div
        className="for_input_track"
      >
          
        <input
          disabled={false}
          onClick={[Function]}
          type="button"
          value="add to my play-list"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`работа Page_All_Music_to_Render 1`] = `
<div
  className="pageContainerOfMainPage"
>
  <div
    className="btn_All_Music"
  >
    <div
      className="input_text"
    >
      <input
        onChange={undefined}
        type="text"
        value=""
      />
      <input
        onClick={undefined}
        type="button"
        value="очистить поиск"
      />
    </div>
    <div
      className="input_btn"
    >
      <input
        onClick={undefined}
        type="button"
        value="показать все"
      />
      <input
        onClick={undefined}
        type="button"
        value="показать недобавленные"
      />
      <input
        onClick={undefined}
        type="button"
        value="показать добавленные"
      />
    </div>
  </div>
  <div
    className="container_track"
  >
    <div
      className="shortMode_track"
    >
      <div
        className="for_span_track"
      >
         
        <span
          className="songN"
        >
          Status Quo
           
          -In The Army Now
        </span>
         
      </div>
      <div
        className="for_audio_track"
      >
           
        <audio
          controls={true}
          src="../songs/Status Quo - In The Army Now.mp3"
        />
      </div>
      <div
        className="for_input_track"
      >
          
        <input
          disabled={false}
          onClick={[Function]}
          type="button"
          value="add to my play-list"
        />
      </div>
    </div>
  </div>
  <div
    className="container_track"
  >
    <div
      className="shortMode_track"
    >
      <div
        className="for_span_track"
      >
         
        <span
          className="songN"
        >
          Stereophonics
           
          -Maybe Tomorrow
        </span>
         
      </div>
      <div
        className="for_audio_track"
      >
           
        <audio
          controls={true}
          src="../songs/Stereophonics - Maybe Tomorrow.mp3"
        />
      </div>
      <div
        className="for_input_track"
      >
          
        <input
          disabled={true}
          onClick={[Function]}
          type="button"
          value="add to my play-list"
        />
      </div>
    </div>
  </div>
  <div
    className="container_track"
  >
    <div
      className="shortMode_track"
    >
      <div
        className="for_span_track"
      >
         
        <span
          className="songN"
        >
          System Of A Dow
           
          -Lonely Day
        </span>
         
      </div>
      <div
        className="for_audio_track"
      >
           
        <audio
          controls={true}
          src="../songs/System Of A Down - Lonely Day.mp3"
        />
      </div>
      <div
        className="for_input_track"
      >
          
        <input
          disabled={false}
          onClick={[Function]}
          type="button"
          value="add to my play-list"
        />
      </div>
    </div>
  </div>
  <div
    className="container_track"
  >
    <div
      className="shortMode_track"
    >
      <div
        className="for_span_track"
      >
         
        <span
          className="songN"
        >
          The Beatles
           
          -Come Together
        </span>
         
      </div>
      <div
        className="for_audio_track"
      >
           
        <audio
          controls={true}
          src="../songs/The Beatles - Come Together.mp3"
        />
      </div>
      <div
        className="for_input_track"
      >
          
        <input
          disabled={false}
          onClick={[Function]}
          type="button"
          value="add to my play-list"
        />
      </div>
    </div>
  </div>
</div>
`;
